src/modules/sensors/vehicle_air_data/VehicleAirData.cpp

```diff
@@ -0,0 +1,280 @@
+{
+	// check if the selected sensor has updated
+	sensor_correction_s corrections;
+
+	if (_sensor_correction_sub.update(&corrections)) {
+		for (int baro_index = 0; baro_index < MAX_SENSOR_COUNT; baro_index++) {
+			const sensor_baro_s &baro = _last_data[baro_index];
+
+			// find sensor (by device id) in sensor_correction
+			if (_sensor_correction_index[baro_index] < 0) {
+				for (int correction_index = 0; correction_index < MAX_SENSOR_COUNT; correction_index++) {
+					if ((baro.device_id > 0) && (corrections.baro_device_ids[correction_index] == baro.device_id)) {
+						_sensor_correction_index[baro_index] = correction_index;
+					}
+				}
+			}
+
+			switch (_sensor_correction_index[baro_index]) {
+			case 0:
+				_offset[baro_index] = corrections.baro_offset_0;
+				_scale[baro_index] = corrections.baro_scale_0;
+				break;
+
+			case 1:
+				_offset[baro_index] = corrections.baro_offset_1;
+				_scale[baro_index] = corrections.baro_scale_1;
+				break;
+
+			case 2:
+				_offset[baro_index] = corrections.baro_offset_2;
+				_scale[baro_index] = corrections.baro_scale_2;
+				break;
+
+			default:
+				_offset[baro_index] = 0.f;
+				_scale[baro_index] = 1.f;
+			}
+		}
+	}
+}
+
+void VehicleAirData::ParametersUpdate()
+{
+	// Check if parameters have changed
+	if (_params_sub.updated()) {
+		// clear update
+		parameter_update_s param_update;
+		_params_sub.copy(&param_update);
+
+		updateParams();
+	}
+}
+
+void VehicleAirData::Run()
+{
+	perf_begin(_cycle_perf);
+
+	bool updated[MAX_SENSOR_COUNT] {};
+
+	for (int uorb_index = 0; uorb_index < MAX_SENSOR_COUNT; uorb_index++) {
+
+		if (!_advertised[uorb_index]) {
+			// use data's timestamp to throttle advertisement checks
+			if (hrt_elapsed_time(&_last_data[uorb_index].timestamp) > 1_s) {
+				if (_sensor_sub[uorb_index].advertised()) {
+					if (uorb_index > 0) {
+						/* the first always exists, but for each further sensor, add a new validator */
+						if (!_voter.add_new_validator()) {
+							PX4_ERR("failed to add validator for sensor_baro:%i", uorb_index);
+						}
+					}
+
+					_advertised[uorb_index] = true;
+
+				} else {
+					_last_data[uorb_index].timestamp = hrt_absolute_time();
+				}
+			}
+
+		} else {
+			updated[uorb_index] = _sensor_sub[uorb_index].update(&_last_data[uorb_index]);
+
+			if (updated[uorb_index]) {
+				if (_priority[uorb_index] == 0) {
+					// set initial priority
+					_priority[uorb_index] = _sensor_sub[uorb_index].get_priority();
+				}
+
+				// pressure corrected with offset and scale (if available)
+				float pressure_corrected = (_last_data[uorb_index].pressure - _offset[uorb_index]) * _scale[uorb_index];
+
+				float vect[3] {pressure_corrected, _last_data[uorb_index].temperature, 0.f};
+				_voter.put(uorb_index, _last_data[uorb_index].timestamp, vect, _last_data[uorb_index].error_count,
+					   _priority[uorb_index]);
+			}
+		}
+	}
+
+	// check for the current best sensor
+	int best_index = 0;
```

Comment:
Should you initialize it at `-1` or is `0` safe?

Reply:
Zero is safe (defaulting to the first baro instance).